docs/guide/static-module-assets.md

```diff
@@ -0,0 +1,45 @@
+# Using static module assets
+
+While developers can place static files in the top-level `/public` folder of the project, which allows a file like `/public/images/bg.png` to be accessed via the URL `https://examplesite.com/images/bg.png`, this isn't a good solution for some situations.
+
+One issue is that the URL does not change with each new deployment, so if `bg.png` has been changed a stale version may be served by the browser's cache.
+
+Another issue is that those who configure Apostrophe to copy CSS and JS assets to Amazon S3 and other CDNs will discover that code like `url("/images/bg.png")` no longer works, because the static asset files are not on the same host with the CSS code. This can be worked around with absolute URLs, but this gives up some of the benefit of using a CDN.
+
+Apostrophe offers a solution to this problem: each module folder can have its own `public` subdirectory, and *these static assets are copied to the same release location* where the CSS and JS bundles are deployed, regardless of whether that is on the server's hard drive, in S3 or some third location.
+
+::: note
+The `public` folder of each module solves a different problem from `ui/public`. While `ui/public` is for JavaScript and CSS files that should be appended as-is to Apostrophe's JavaScript and CSS bundles, usually to accommodate a custom webpack build, `public` is for files that should be **available separately.** A common example is a `.png` file to be used as a background image in CSS.
+:::
+
+#### Automatic asset path correction in stylesheets
+
+Files you place in the `public` subdirectory of any module are always deployed such that you can write URLs like this in your CSS or `.scss` files:
+
+`/modules/mymodulename/images/bg.png`
+
+Apostrophe will automatically fix these "asset paths" so they refer to the final URL of the asset, no matter what your production environment looks like.
+
+In addition, in production deployments, the final URL will always contain a "release identifier" so that any static assets in the browser cache from a previous release are not reused.
+
+#### Asset paths in Nunjucks templates
+
+You can also convert asset paths to URLs in a Nunjucks template by calling the `apos.asset.url` helper function, like this:
+
+```django
+{{ apos.asset.url('/modules/mymodulename/images/bg.png') }}
+```
+
+#### Asset paths in frontend JavaScript
+
+You can obtain the URL corresponding to an asset path in frontend JavaScript using `apos.util.assetUrl`:
```

Comment:
"Finally, you can obtain..."

Reply:
Based on earlier discussions we're going with "front end." There's a little note about that on the contribution page.